src/foam/core/reflow/Console.js

```diff
@@ -1547,6 +1581,21 @@ foam.CLASS({
       delay: 500,
       code: function() {
         this.maybeRegenScript();
+        this.saveScriptToLocalStorage();
+      }
+    },
+    {
+      name: 'saveScriptToLocalStorage',
+      isMerged: true,
+      mergeDelay: 1000,
+      code: function() {
+        if ( this.value && this.value.script ) {
+          var autosaveData = {
+            script: this.value.script,
+            name: this.value.name
+          };
+          this.window.localStorage[this.AUTOSAVED_SCRIPT] = JSON.stringify(autosaveData);
```

Comment:
Shouldn't this include the script name as part of the key? Otherwise, if you have multiple tabs open they will keep overwriting each other?

Reply:
done.